src/PSR7/Validators/QueryArgumentsValidator.php

```diff
@@ -108,4 +126,21 @@ private function parseQueryArguments(RequestInterface $message) : array
 
         return $parsedQueryArguments;
     }
+
+    /**
+     * @return mixed
+     *
+     * @throws InvalidQueryArgs
+     */
+    private static function parseArgumentValue(string $value, string $contentType, OperationAddress $addr, string $argumentName)
+    {
+        if (preg_match('#^application/.*json$#', $contentType)) {
+            $value = json_decode($value, true);
+            if (json_last_error() !== JSON_ERROR_NONE) {
+                throw InvalidQueryArgs::becauseValueIsNotValidJson(json_last_error_msg(), $argumentName, $addr);
```

Comment:
How about more generic `becauseDoesNotMatchContentType` or something like this? We can reuse it later for XML or other types